samples/extensions/extended_dynamic_state2/extended_dynamic_state2.h

```diff
@@ -0,0 +1,169 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+#include "scene_graph/components/pbr_material.h"
+
+class ExtendedDynamicState2 : public ApiVulkanSample
+{
+  public:
+	typedef struct ModelDynamicParam
+	{
+		bool depth_bias         = false;
+		bool rasterizer_discard = false;
+	} ModelDynamicParam;
+
+	struct
+	{
+		bool                           tessellation = false;
+		float                          tess_factor  = 1.0;
+		float                          patch_control_points_float{3.0f};
+		int                            patch_control_points{3};
+		std::vector<ModelDynamicParam> objects;
+		int                            selected_obj     = 0;
+		bool                           selection_active = true;
+		bool                           time_tick        = false;
+	} gui_settings;
+
+	struct UBOBAS
+	{
+		glm::mat4 projection;
+		glm::mat4 view;
+		glm::vec4 ambientLightColor = glm::vec4(1.f, 1.f, 1.f, 0.1f);
+		glm::vec4 lightPosition     = glm::vec4(-3.0f, -8.0f, 6.0f, -1.0f);
+		glm::vec4 lightColor        = glm::vec4(1.0f, 1.0f, 1.0f, 1.0f);
+		float     lightIntensity    = 50.0f;
+	} ubo_baseline;
+
+	struct UBOTESS
+	{
+		glm::mat4 projection;
+		glm::mat4 modelview;
+		float     tessellation_factor = 1.0f;
+	} ubo_tess;
+
+	struct UBOBG
+	{
+		glm::mat4 projection;
+		glm::mat4 background_modelview;
+
+	} ubo_background;
+
+	VkDescriptorPool descriptor_pool{VK_NULL_HANDLE};
+
+	struct
+	{
+		VkDescriptorSetLayout baseline{VK_NULL_HANDLE};
+		VkDescriptorSetLayout tesselation{VK_NULL_HANDLE};
+		VkDescriptorSetLayout background{VK_NULL_HANDLE};
+	} descriptor_set_layouts;
+
+	struct
+	{
+		VkPipelineLayout baseline{VK_NULL_HANDLE};
+		VkPipelineLayout tesselation{VK_NULL_HANDLE};
+		VkPipelineLayout background{VK_NULL_HANDLE};
+	} pipeline_layouts;
+
+	struct
+	{
+		VkDescriptorSet baseline{VK_NULL_HANDLE};
+		VkDescriptorSet tesselation{VK_NULL_HANDLE};
+		VkDescriptorSet background{VK_NULL_HANDLE};
+	} descriptor_sets;
+
+	struct
+	{
+		VkPipeline baseline{VK_NULL_HANDLE};
+		VkPipeline tesselation{VK_NULL_HANDLE};
+		VkPipeline background{VK_NULL_HANDLE};
+	} pipeline;
+
+	struct
+	{
+		std::unique_ptr<vkb::core::Buffer> baseline;
+		std::unique_ptr<vkb::core::Buffer> tesselation;
+		std::unique_ptr<vkb::core::Buffer> background;
+	} uniform_buffers;
+
+	struct
+	{
+		Texture envmap;
+	} textures;
+
+	struct
+	{
+		glm::mat4 model_matrix;
+		glm::vec4 color;
+	} push_const_block;
+
+	std::unique_ptr<vkb::sg::Scene> scene;
+	struct SceneNode
+	{
+		std::string       name;
+		vkb::sg::Node    *node;
+		vkb::sg::SubMesh *sub_mesh;
+	};
+	std::vector<std::vector<SceneNode>> scene_nodes;
```

Comment:
You never need the the fact that your three scenes are in a vector.
Maybe just having three separate `std::vector<SceneNode>`s would be easier to read?

Reply:
fixed